src/assignments/mars-rover.rst

```diff
@@ -185,21 +267,102 @@ several functions outside of ``constructor``.
+          object that corresponds to one ``Command`` in ``message.commands``.
+         
+   c. Updates certain properties of the rover object
+
+      i. Details about how to respond to different commands are in the
+         :ref:`Command Types table <command-types-table>`.
 
 .. admonition:: Example
 
    .. sourcecode:: js
 
       let commands = [new Command('MODE_CHANGE', 'LOW_POWER'), new Command('STATUS_CHECK')];
-      let message = new Message('e1', commands);
-      let rover = new Rover(98382);
+      let message = new Message('Test message with two commands', commands);
+      let rover = new Rover(98382);    // Passes 98382 as the rover's position.
       let response = rover.receiveMessage(message);
 
+      console.log(response);
+
+   **Output**
+
+   ::
+
+      {
+         message: 'Test message with two commands',
+         results: [
+            {
+               completed: true
+            },
+            {
+               completed: true, 
+               roverStatus: {
+                  mode: 'LOW_POWER', 
+                  generatorWatts: 110, 
+                  position: 98382
+               }
+            }
+         ]
+      }
+
+
+``Rover`` Tests
+^^^^^^^^^^^^^^^
+
+Create ``spec/rover.spec.js`` and write the following tests. Write the code to
+make them pass in ``rover.js``. Remember to use the given phrase as the test
+description.
+
+Test 7 
+   "constructor sets position and default values for mode and generatorWatts".
+   Refer to the :ref:`Rover Class <rover-class>` description above for these
+   default values.
+
+Test 8
+   "response returned by receiveMessage contains name of message"
+
+Test 9
+   "response returned by receiveMessage includes two results if two commands
+   are sent in the message"
+
+Test 10
+   "responds correctly to status check command"
+
+   a. For the ``STATUS_CHECK`` command, ``receiveMessage(message).results`` 
+      includes a ``roverStatus`` object describing the current state of the 
+      rover object --- ``mode``, ``generatorWatts``, and ``position``. The test 
+      should check each of these for accuracy.
+   b. See the :ref:`Rover Command Types <command-types-table>` table for more
+      details.
+
+Test 11
+   "responds correctly to mode change command". 
+   
+   a. The test should check the ``completed`` property and rover mode for accuracy.
+   b. The rover has two modes that can be passed a values to a mode change command,
+      'LOW_POWER' and 'NORMAL'.
+
+Test 12
+   "responds with false completed value when attempting to move in LOW_POWER
+   mode". 
+   
+   a. The test should check the ``completed`` property for accuracy and confirm 
+      that the rover position did not change.
+   b. Use the :ref:`Rover Modes table <rover-modes-table>` for guidance on how
+      to handle move commands in different modes.
+
+Test 13
+   "responds with position for move command".
+
+   a. A move command will update the rover's position with the position value in 
```

Comment:
Slight rephrasing: "A ``MOVE`` command will update the rover's position to the value sent with the command."